Dockerfile

```diff
@@ -19,39 +18,32 @@ RUN bin/build-ui.sh
 
 
 # Next, all the things for building the Python web service
-
 FROM python:3.7-slim@sha256:3e4be41076ebb6fe8c3112b220ce133ef0dc49c814024e4874ca76eae3c8dec0
 
+WORKDIR /app/
+
 ENV PYTHONUNBUFFERED=1 \
     PYTHONPATH=/app/
 
 # add a non-privileged user for installing and running the application
 # don't use --create-home option to prevent populating with skeleton files
-RUN mkdir /app && \
-    chown 10001:10001 /app && \
-    groupadd --gid 10001 app && \
+RUN groupadd --gid 10001 app && \
     useradd --no-create-home --uid 10001 --gid 10001 --home-dir /app app
 
-COPY . /app
-
 # Install Python dependencies
-COPY requirements.txt /tmp/
-WORKDIR /tmp
+COPY ./requirements.txt /app/requirements.txt
 RUN pip install --no-cache-dir -r requirements.txt
 
-# COPY . /app
-
-# Switch back to home directory
-WORKDIR /app
+# Copy the app
+COPY . /app
 
 # Copy built static assets from the ui container
 COPY --from=ui /app/ui/build /app/ui/build
 
-RUN chown -R 10001:10001 /app
-
-USER 10001
+# app should own everything under /app in the container
+RUN chown -R app.app /app
```

Comment:
Using `app` rather than `10001` allows us to fiddle with the uid/gid without having to update all the places we use the id.